### PR TITLE
*: add feature "push"

### LINKED ISSUE #84: Allow building without hyper
In the ideal use of Prometheus, libraries provide their own instrumentation:

> The short answer is to instrument everything. Every library, subsystem and service should have at least a few metrics to give you a rough idea of how it is performing.
―https://prometheus.io/docs/practices/instrumentation/

Bringing in the push client by default makes that hard to do: any library instrumented with Prometheus suddenly brings in hyper and all its dependencies.

Here are a couple of ideas on how to fix this:

- make the push gateway client a separate crate
- add an optional feature to enable the push gateway client

### PR DESCRIPTION
**Pull request by overvenus**
This PR introduces a new feature `push`.  It allows building core instrumentation library without hyper and libc.

~~There are a few API changes:~~
  - ~~mod `push` is published.~~
  - ~~following functions are moved to mod `push`.~~
    ```
    push_metrics
    push_add_metrics
    push_collector
    push_add_collector
    hostname_grouping_key
    ```

Close #84 

PTAL @siddontang @kamalmarhubi

### CODE REVIEW

File: examples/example_push.rs [hunk 1]
```diff
@@ -17,59 +17,78 @@ extern crate prometheus;
 extern crate lazy_static;
 extern crate getopts;
 
-use std::env;
-use std::time;
-use std::thread;
+mod bridge {
```
Comment (kamalmarhubi) [inline examples/example_push.rs]: wow it's even nicer than I thought it would be after that change! :-)

**Comment by overvenus** [comment]
PTAL @siddontang

**Comment by siddontang** [comment]
Do we need to update TiKV? 
Please update the cargo version too.

**Comment by siddontang** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

File: .travis.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@ addons:
 env:
   global:
     - RUSTFLAGS=--deny=warnings
+    - FEATURES=push
```
Comment (BusyJay) [inline .travis.yml]: Should test whether everything works fine when the features is disabled.

**Comment by siddontang** [comment]
PTAL @BusyJay

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**